### PR TITLE
DAT-3786 | Infobox Builder: add dialog on onbeforeunload

### PR DESCRIPTION
**Pull request by rybmat**
## Links

* https://wikia-inc.atlassian.net/browse/DAT-3786

## Description

Handling onbeforeunload event to prevent user from accidentally loosing changes.

## Reviewers
@Wikia/x-wing 
@dianafa

### CODE REVIEW

File: front/main/app/routes/infobox-builder.js [hunk 1]
```diff
@@ -294,6 +306,7 @@ export default Ember.Route.extend({
 		save() {
 			const model = this.modelFor('infoboxBuilder');
 
+			window.onbeforeunload = null;
```
Comment (vforge) [inline front/main/app/routes/infobox-builder.js]: It would be better to revert to previous value of `window.onbeforeunload` instead of setting null.

**Comment by rogatty** [comment]
How about using https://github.com/jasonmit/ember-onbeforeunload?
It provides a mixin that has more sophisticated logic for doing that: https://github.com/jasonmit/ember-onbeforeunload/blob/master/addon/mixins/confirmation.js

File: front/main/app/routes/infobox-builder.js [hunk 2]
```diff
@@ -161,6 +162,17 @@ export default Ember.Route.extend({
 	},
 
 	/**
+	 * @desc setups handlers for events
+	 *
+	 * @returns {void}
+	 */
+	setupEventHandlers() {
+		window.onbeforeunload = function () {
```
Comment (Warkot) [inline front/main/app/routes/infobox-builder.js]: Wouldn't it be possible to use the fat arrow shorthand here like this?
```
window.onbeforeunload = () => i18n.t('infobox-builder:main.leave-confirmation');
```

Comment (rybmat) [inline front/main/app/routes/infobox-builder.js]: it should be possible, but I'm going to change it to what Igor proposed

Comment (Warkot) [inline front/main/app/routes/infobox-builder.js]: :+1:

File: front/main/app/controllers/infobox-builder.js [hunk 3]
```diff
@@ -24,6 +27,9 @@ export default Ember.Controller.extend({
 		save() {
 			const model = this.get('model');
 
+			// prevents showing confirmation dialog on save
+			this.isDirty = false;
```
Comment (rogatty) [inline front/main/app/controllers/infobox-builder.js]: ember-onbeforeunload uses getter (https://github.com/jasonmit/ember-onbeforeunload/blob/master/addon/mixins/confirmation.js#L23) so please use setter everywhere (`this.set('isDirty', false)`).

File: front/main/app/controllers/infobox-builder.js [hunk 4]
```diff
@@ -1,6 +1,9 @@
 import Ember from 'ember';
 
 export default Ember.Controller.extend({
+
+	isDirty: false,
```
Comment (rogatty) [inline front/main/app/controllers/infobox-builder.js]: Please add a comment that this is used by ember-onbeforeunload.

File: front/main/app/components/infobox-builder-wrapper.js [hunk 5]
```diff
@@ -3,3 +3,4 @@ import Ember from 'ember';
 export default Ember.Component.extend({
 	classNames: ['infobox-builder-preview']
 });
+
```
Comment (rogatty) [inline front/main/app/components/infobox-builder-wrapper.js]: No need for this empty line.

**Comment by rogatty** [comment]
LGTM

**Pull request merged**